src/Diff.ml

```diff
@@ -18,18 +21,21 @@ module MakeDiff (IO1 : ImageIO.ImageIO) (IO2 : ImageIO.ImageIO) = struct
 
   let compare (base : IO1.t ImageIO.img) (comp : IO2.t ImageIO.img)
       ?(antialiasing = false) ?(outputDiffMask = false) ?(diffLines = false)
-      ?(diffPixel : int * int * int = redPixel) ?(threshold = 0.1)
-      ?(ignoreRegions = []) () =
+      ?diffPixel ?(threshold = 0.1) ?(ignoreRegions = []) () =
     let maxDelta = maxYIQPossibleDelta *. (threshold ** 2.) in
+    let diffPixel = match diffPixel with Some x -> x | None -> redPixel in
     let diffOutput =
       match outputDiffMask with
       | true -> IO1.makeSameAsLayout base
       | false -> base
     in
-    let diffPixelQueue = Queue.create () in
+
+    let diffCount = ref 0 in
     let diffLinesStack = Stack.create () in
     let countDifference x y =
-      diffPixelQueue |> Queue.push (x, y);
+      incr diffCount;
+      IO1.setImgColor ~x ~y diffPixel diffOutput;
```

Comment:
@dmtrKovalenko 
Setting the diff color directly can break the antialiasing detection in some cases. 

When diffMask is off, we are changing the pixel color on the base image directly, which we might use later to determine antialiased pixels. 
(Remember, that we look at the surrounding 8 pixels to detect antialiasing). So we do look at "past" pixels, which with this change might not be the original color anymore.

Reply:
But the output image is a totally new buffer copied from the original image and not the same source. 

This queue is having a lot of memory overhead for every run

Reply:
I think on line 30 its the same buffer we use for comparison. We only create a new buffer, when "outputDiffMask" is true. 
I might be wrong though. Its been a long time since i looked at this code 😄

Reply:
Thanks need to check this again